library-udf/src/test/java/org/apache/iotdb/library/dmatch/DMatchTests.java

```diff
@@ -0,0 +1,782 @@
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      String result1 = resultSet.getString(1);
+      resultSet.next();
+      String result2 = resultSet.getString(1);
+      resultSet.next();
+      String result3 = resultSet.getString(1);
+      resultSet.next();
+      String result4 = resultSet.getString(1);
+      resultSet.next();
+      String result5 = resultSet.getString(1);
+      resultSet.next();
+      String result6 = resultSet.getString(1);
+      resultSet.next();
+      String result7 = resultSet.getString(1);
+      resultSet.next();
+      String result8 = resultSet.getString(1);
+      resultSet.next();
+      String result9 = resultSet.getString(1);
+      Assert.assertEquals("0.0", result1);
+      Assert.assertEquals("4.0", result2);
+      Assert.assertEquals("9.6", result3);
+      Assert.assertEquals("13.4", result4);
+      Assert.assertEquals("20.0", result5);
+      Assert.assertEquals("15.6", result6);
+      Assert.assertEquals("9.2", result7);
+      Assert.assertEquals("11.8", result8);
+      Assert.assertEquals("6.0", result9);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testXCorr3() {
+    String sqlStr = "select xcorr(d4.s3, d4.s7) from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      String result1 = resultSet.getString(1);
+      resultSet.next();
+      String result2 = resultSet.getString(1);
+      resultSet.next();
+      String result3 = resultSet.getString(1);
+      resultSet.next();
+      String result4 = resultSet.getString(1);
+      resultSet.next();
+      String result5 = resultSet.getString(1);
+      resultSet.next();
+      String result6 = resultSet.getString(1);
+      resultSet.next();
+      String result7 = resultSet.getString(1);
+      resultSet.next();
+      String result8 = resultSet.getString(1);
+      resultSet.next();
+      String result9 = resultSet.getString(1);
+      Assert.assertEquals("0.0", result1);
+      Assert.assertEquals("4.0", result2);
+      Assert.assertEquals("9.6", result3);
+      Assert.assertEquals("13.4", result4);
+      Assert.assertEquals("20.0", result5);
+      Assert.assertEquals("15.6", result6);
+      Assert.assertEquals("9.2", result7);
+      Assert.assertEquals("11.8", result8);
+      Assert.assertEquals("6.0", result9);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testXCorr4() {
+    String sqlStr = "select xcorr(d4.s4,d4.s8) from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      String result1 = resultSet.getString(1);
```

Comment:
Missing `resultSet.next()` before this line?

Reply:
Done.